# Get non-repeat peaks
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort.narrowPeak.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_repeated.bed --peakNameCol 3 --removePeaks --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed

# Get non-repeat enhancers whose orthologs are closed in at least 1 other species
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_repeated.bed --peakNameCol 3 --removePeaks --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_noRepeats.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_noRepeats.bed

# Get non-repeat enhancers whose orthologs are closed in rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_repeated.bed --peakNameCol 3 --removePeaks --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats.bed

# Get all test set enhancers in macaque, rat, and bat
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_andRat_andBat.fa

# Get all test set mouse non-enhancer orthologs in macaque, rat, and bat
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.plusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.plusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_summitPlusMinus250bp_andrn6_andHLrouAeg4.fa

# Get macaque enhancers whose mouse orthologs are not enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed

# Get macaque non-enhancers whose mouse orthologs are enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed

# Get rat enhancers whose mouse orthologs are not enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test.bed

# Get rat non-enhancers whose mouse orthologs are enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed

# Get bat enhancers whose mouse orthologs are not enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed

# Get bat non-enhancers whose mouse orthologs are enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed

# Get non-repeat glires-specific enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_repeated.bed --peakNameCol 3 --removePeaks --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed

# Get euarchonta-specific enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_notInRat_notInBat.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed

# Get euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_inBatLoose.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed

# Get enhancers that are euarchonta-specific relative to glires
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_notInRat.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed

# Get non-enhancers that are euarchonta-specific relative to glires
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed

# Get bat-specific enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_notInMacaque.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_test.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rn6_summitExtendedMin50Max2XProtect5_notInRat.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed

# Get bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_inMacaqueLoose.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_test.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_inRatLoose.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed # Only 39 peaks [not using]
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames.bed -b /projects/MPRA/Irene/macaque/atac-pipeline-output/OfM/peak/pooled-rep/basename_prefix.pooled.pval0.01.300K.bfilt.narrowPeak.gz | intersectBed -wa -u -a stdin -b /projects/MPRA/Irene/macaque/atac-pipeline-output/M1/peak/pooled-rep/basename_prefix.pooled.pval0.01.300K.bfilt.narrowPeak.gz > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_inMacaqueLoose.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_inMacaqueLoose.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_inMacaqueLoose_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed # Peak names are NOT reliable
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed

# Get non-repeat cortex enhancers that are also striatum enhancers
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort.bed | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum.bed.gz

# Get striatum enhancers that are not cortex enhancers
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed.narrowPeak.gz -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_AgeB_ATAC_out/peak/macs2/pooled_rep/Cortex_ATAC_AgeB_Rep1_techRepsPooled.nodup.tn5_pooled.pf.pval0.01.300K.filt.narrowPeak.gz | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_AgeC_ATAC_out/peak/macs2/idr/optimal_set/Cortex_AgeC_ATAC_out_ppr.IDR0.1.filt.narrowPeak.gz | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex.gz

# Get non-repeat liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver.narrowPeak.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_repeated.bed --peakNameCol 3 --removePeaks --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats.bed

# Get macaque cortex enhancers that overlap striatum enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort.bed | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test.bed.gz

# Get macaque striatum enhancers that are not cortex enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2Protect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2Protect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed

# Get macaque cortex enhancers that overlap liver enhancers
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort.bed | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test.bed.gz

# Get macaque liver enhancers that are not cortex enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test.bed
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/MPRA/Irene/macaque/atac-pipeline-output/OfM/peak/pooled-rep/basename_prefix.pooled.pval0.01.300K.bfilt.narrowPeak.gz | subtractBed -A -a stdin -b /projects/MPRA/Irene/macaque/atac-pipeline-output/M1/peak/pooled-rep/basename_prefix.pooled.pval0.01.300K.bfilt.narrowPeak.gz | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz

# Get rat cortex enhancers that overlap rat striatum enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test.bed.gz

# Get rat striatum enhancers that are not rat cortex enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed

# Get rat cortex enhancers that overlap rat liver enhancers
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed.gz -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.bed.gz | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test.bed.gz

# Get rat liver enhancers that are not rat cortex enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/MPRA/Irene/rats/atac-pipeline-output/M1/peak/pooled-rep/basename_prefix.pooled.pval0.01.300K.bfilt.narrowPeak.gz | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz

# Get bat cortex enhancers that overlap bat striatum enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort.bed.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort.bed | gzip > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test.bed.gz

# Get bat striatum enhancers that are not bat cortex enhancers
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrCLoose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed

# Evaluate performance on test set for negative set with non-enhancer orthologs of enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_andNeg_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_predictedProba.txt --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.454954, accuracy: 0.792858, sensitivity: 0.777129, specificity: 0.817007, auc: 0.877842, precision: 0.867022, NPV: 0.704816, auprc: 0.822464, aunpvspec: 0.914656, auprg: 0.820860, F1: 0.819618, recallAt0.05FDR: 0.142017, recallAt0.1FDR: 0.364732
# Performance on mouse enhancers whose orthologs are not cortex enhancers in at least one other species and negative set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBatNonLoose.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_test_andNeg_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_predictedProba.txt --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4832
# Negatives: 4620
# Performance: loss: 0.456249, accuracy: 0.792478, sensitivity: 0.775869, specificity: 0.809848, auc: 0.875288, precision: 0.810157, NPV: 0.775521, auprc: 0.866755, auprg: 0.777076, F1: 0.792642, recallAt0.05FDR: 0.269805, recallAt0.1FDR: 0.489069
# Performance on mouse enhancers whose orthologs are not cortex enhancers in rat and non-enhancer orthologs of rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2779
# Negatives: 2064
# Performance: loss: 0.483077, accuracy: 0.777824, sensitivity: 0.783195, specificity: 0.770591, auc: 0.858119, precision: 0.821321, NPV: 0.725262, auprc: 0.809332, auprg: 0.770665, F1: 0.801805, recallAt0.05FDR: 0.070979, recallAt0.1FDR: 0.305959
# Performance on all enhancers from macaque, rat, and bat and mouse non-enhancer orthologs in macaque, rat, and bat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_summitPlusMinus250bp_andrn6_andHLrouAeg4.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 12858
# Negatives: 7977
# Performance: loss: 0.541488, accuracy: 0.755316, sensitivity: 0.751205, specificity: 0.761941, auc: 0.832461, precision: 0.835698, NPV: 0.655169, auprc: 0.766648, aunpvspec: 0.878085, auprg: 0.752912, F1: 0.791202, recallAt0.05FDR: 0.075530, recallAt0.1FDR: 0.226589
# Performance on macaque enhancers whose orthologs are not cortex enhancers in mouse and negative set corresponding to non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1664
# Negatives: 2625
# Performance: loss: 0.455091, accuracy: 0.801189, sensitivity: 0.771935, specificity: 0.819741, auc: 0.877432, precision: 0.730868, NPV: 0.850030, auprc: 0.813499, aunpvspec: 0.916622, auprg: 0.820007, F1: 0.750840, recallAt0.05FDR: 0.134315, recallAt0.1FDR: 0.308894
# Performance on rat enhancers whose orthologs are not cortex enhancers in mouse and rat non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2064
# Negatives: 2779
# Performance: loss: 0.739640, accuracy: 0.666323, sensitivity: 0.671996, specificity: 0.662109, auc: 0.727621, precision: 0.596303, NPV: 0.731029, auprc: 0.653692, aunpvspec: 0.774956, auprg: 0.499413, F1: 0.631891, recallAt0.05FDR: 0.006056, recallAt0.1FDR: 0.009690
# Performance on bat enhancers whose orthologs are not cortex enhancers in mouse and bat non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --chromSizesFileName /projects/MPRA/Simone/Bats/Bat1K/mRouAeg1.chrom.sizes --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 729
# Negatives: 2574
# Performance: loss: 0.437635, accuracy: 0.816530, sensitivity: 0.836763, specificity: 0.810800, auc: 0.899307, precision: 0.556062, NPV: 0.946056, auprc: 0.701104, aunpvspec: 0.969766, auprg: 0.892049, F1: 0.668127, recallAt0.05FDR: 0.019890, recallAt0.1FDR: 0.049383
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 627
# Negatives: 271
# Performance: loss: 0.421140, accuracy: 0.809577, sensitivity: 0.791866, specificity: 0.850554, auc: 0.899807, precision: 0.924581, NPV: 0.638504, auprc: 0.807049, aunpvspec: 0.950812, auprg: 0.888122, F1: 0.853093, recallAt0.05FDR: 0.228782, recallAt0.1FDR: 0.350554
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.539095, accuracy: 0.754510, sensitivity: 0.776910, specificity: 0.690000, auc: 0.806337, precision: 0.878312, NPV: 0.517824, auprc: 0.635652, aunpvspec: 0.908861, F1: 0.824505, recallAt0.05FDR: 0.032500, recallAt0.1FDR: 0.047500
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.497850, accuracy: 0.776969, sensitivity: 0.775977, specificity: 0.778023, auc: 0.856811, precision: 0.787764, NPV: 0.765856, auprc: 0.855163, aunpvspec: 0.849879, auprg: 0.749205, F1: 0.781826, recallAt0.05FDR: 0.250221, recallAt0.1FDR: 0.478817
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.589287, accuracy: 0.758166, sensitivity: 0.826316, specificity: 0.720157, auc: 0.838130, precision: 0.622193, NPV: 0.881437, auprc: 0.704652, aunpvspec: 0.910827, auprg: 0.727126, F1: 0.709872, recallAt0.05FDR: 0.028070, recallAt0.1FDR: 0.082456
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3067
# Negatives: 3761
# Performance: loss: 0.690954, accuracy: 0.694640, sensitivity: 0.769482, specificity: 0.633608, auc: 0.775727, precision: 0.631354, NPV: 0.771197, auprc: 0.746957, aunpvspec: 0.781207, auprg: 0.605633, F1: 0.693608, recallAt0.05FDR: 0.044506, recallAt0.1FDR: 0.128301
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver.narrowPeak.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 684
# Negatives: 1968
# Performance: loss: 0.334643, accuracy: 0.867685, sensitivity: 0.786382, specificity: 0.898628, auc: 0.915813, precision: 0.746988, NPV: 0.917034, auprc: 0.834725, aunpvspec: 0.959043, auprg: 0.928530, F1: 0.766179, recallAt0.05FDR: 0.305741, recallAt0.1FDR: 0.450601
# Performance on shared macaque cortex and caudate-putamen enhancers and macaque caudate-putamen enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1769
# Negatives: 1081
# Performance: loss: 0.640486, accuracy: 0.715789, sensitivity: 0.790560, specificity: 0.593432, auc: 0.763000, precision: 0.760881, NPV: 0.633893, auprc: 0.650992, aunpvspec: 0.834170, auprg: 0.605989, F1: 0.775437, recallAt0.05FDR: 0.005088, recallAt0.1FDR: 0.008326
# Performance on shared macaque cortex and liver enhancers and macaque liver enhancers that are not cortex enhancers 
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 657
# Negatives: 2624
# Performance: loss: 0.345851, accuracy: 0.858947, sensitivity: 0.844749, specificity: 0.866452, auc: 0.925546, precision: 0.769764, NPV: 0.913486, auprc: 0.862012, aunpvspec: 0.959277, auprg: 0.906300, F1: 0.805515, recallAt0.05FDR: 0.164384, recallAt0.1FDR: 0.420091
# Performance on shared rat cortex and striatum enhancers and rat striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 935
# Performance: loss: 0.757216, accuracy: 0.662493, sensitivity: 0.696422, specificity: 0.563102, auc: 0.674043, precision: 0.823618, NPV: 0.387703, auprc: 0.381469, aunpvspec: 0.854495, auprg: 0.449752, F1: 0.754698, recallAt0.05FDR: 0.000000, recallAt0.1FDR: 0.000000
# Performance on rat cortex and liver enhancers and rat liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1136
# Negatives: 1367
# Performance: loss: 0.438247, accuracy: 0.813823, sensitivity: 0.715669, specificity: 0.895391, auc: 0.884518, precision: 0.850418, NPV: 0.791209, auprc: 0.878646, aunpvspec: 0.888927, auprg: 0.829220, F1: 0.777247, recallAt0.05FDR: 0.401849, recallAt0.1FDR: 0.612236
# Performance on shared bat cortex and striatum enhancers and bat striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhLoosePeakLessThan1kb500bp_data_train.KerasModels/hybrid_cortexEnhancerShort_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 886
# Negatives: 746
# Performance: loss: 0.628141, accuracy: 0.723958, sensitivity: 0.826749, specificity: 0.601877, auc: 0.788062, precision: 0.711510, NPV: 0.745228, auprc: 0.753016, aunpvspec: 0.801051, auprg: 0.624144, F1: 0.764813, recallAt0.05FDR: 0.056300, recallAt0.1FDR: 0.084450

# Evaluate performance on test set for negative set with 10X G/C-matched regions and enhancers from other brain regions (striatum, cerebellum, and retina)
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseLiver_andPancreas.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_andNeg_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_predictedProba.txt --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 14505
# Performance: loss: 0.264657, accuracy: 0.890884, sensitivity: 0.779452, specificity: 0.952051, auc: 0.958728, precision: 0.899225, NPV: 0.887186, auprc: 0.925806, aunpvspec: 0.976979, auprg: 0.956133, F1: 0.835066, recallAt0.05FDR: 0.587478, recallAt0.1FDR: 0.777694
# Performance on mouse enhancers whose orthologs are not cortex enhancers in at least one other species and negative set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBatNonLoose.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonCortexOtherSpecies_test_andNeg_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_predictedProba.txt --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4832
# Negatives: 4620
# Performance: loss: 0.470895, accuracy: 0.791208, sensitivity: 0.779594, specificity: 0.803355, auc: 0.877585, precision: 0.805689, NPV: 0.777033, auprc: 0.874146, aunpvspec: 0.878522, auprg: 0.784738, F1: 0.792427, recallAt0.05FDR: 0.329545, recallAt0.1FDR: 0.534199
# Performance on mouse enhancers whose orthologs are not cortex enhancers in rat and non-enhancer orthologs of rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2779
# Negatives: 2064
# Performance: loss: 0.506217, accuracy: 0.776791, sensitivity: 0.787154, specificity: 0.762839, auc: 0.857514, precision: 0.817146, NPV: 0.726916, auprc: 0.812118, aunpvspec: 0.885783, auprg: 0.774511, F1: 0.801870, recallAt0.05FDR: 0.030039, recallAt0.1FDR: 0.310562
# Performance on all enhancers from macaque, rat, and bat and mouse non-enhancer orthologs in macaque, rat, and bat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_summitPlusMinus250bp_andrn6_andHLrouAeg4.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 12858
# Negatives: 7977
# Performance: loss: 0.542979, accuracy: 0.768898, sensitivity: 0.760772, specificity: 0.781998, auc: 0.852543, precision: 0.849058, NPV: 0.669744, auprc: 0.801314, aunpvspec: 0.890687, auprg: 0.796309, F1: 0.802494, recallAt0.05FDR: 0.208537, recallAt0.1FDR: 0.346935
# Performance on macaque enhancers whose orthologs are not cortex enhancers in mouse and negative set corresponding to non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1664
# Negatives: 2625
# Performance: loss: 0.436589, accuracy: 0.813666, sensitivity: 0.792067, specificity: 0.827363, auc: 0.891885, precision: 0.744212, NPV: 0.862535, auprc: 0.827387, aunpvspec: 0.930679, auprg: 0.840398, F1: 0.767394, recallAt0.05FDR: 0.092248, recallAt0.1FDR: 0.338642
# Performance on rat enhancers whose orthologs are not cortex enhancers in mouse and rat non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2064
# Negatives: 2779
# Performance: loss: 0.760235, accuracy: 0.672414, sensitivity: 0.675388, specificity: 0.670205, auc: 0.738773, precision: 0.603333, NPV: 0.735439, auprc: 0.659048, aunpvspec: 0.788128, auprg: 0.522387, F1: 0.637330, recallAt0.05FDR: 0.001211, recallAt0.1FDR: 0.001211
# Performance on bat enhancers whose orthologs are not cortex enhancers in mouse and bat non-enhancer orthologs of mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --chromSizesFileName /projects/MPRA/Simone/Bats/Bat1K/mRouAeg1.chrom.sizes --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 729
# Negatives: 2574
# Performance: loss: 0.353292, accuracy: 0.850893, sensitivity: 0.831276, specificity: 0.856449, auc: 0.923274, precision: 0.621220, NPV: 0.947154, auprc: 0.755120, aunpvspec: 0.977429, auprg: 0.923312, F1: 0.711059, recallAt0.05FDR: 0.017833, recallAt0.1FDR: 0.057613
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 627
# Negatives: 271
# Performance: loss: 0.450101, accuracy: 0.798998, sensitivity: 0.777512, specificity: 0.848708, auc: 0.897643, precision: 0.922422, NPV: 0.622463, auprc: 0.818696, aunpvspec: 0.946121, auprg: 0.897751, F1: 0.843791, recallAt0.05FDR: 0.265683, recallAt0.1FDR: 0.405904
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.480517, accuracy: 0.795163, sensitivity: 0.791771, specificity: 0.798764, auc: 0.874929, precision: 0.806861, NPV: 0.783211, auprc: 0.879400, aunpvspec: 0.863959, auprg: 0.795301, F1: 0.799245, recallAt0.05FDR: 0.380847, recallAt0.1FDR: 0.605914
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.437314, accuracy: 0.805276, sensitivity: 0.833333, specificity: 0.789628, auc: 0.897545, precision: 0.688406, NPV: 0.894678, auprc: 0.801515, aunpvspec: 0.944817, auprg: 0.857259, F1: 0.753968, recallAt0.05FDR: 0.008772, recallAt0.1FDR: 0.189474
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3067
# Negatives: 3761
# Performance: loss: 0.483709, accuracy: 0.793864, sensitivity: 0.741604, specificity: 0.836480, auc: 0.869340, precision: 0.787160, NPV: 0.798781, auprc: 0.855800, aunpvspec: 0.875971, auprg: 0.798548, F1: 0.763704, recallAt0.05FDR: 0.322791, recallAt0.1FDR: 0.526247
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver.narrowPeak.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 684
# Negatives: 1968
# Performance: loss: 0.303361, accuracy: 0.878358, sensitivity: 0.787717, specificity: 0.912856, auc: 0.927589, precision: 0.774787, NPV: 0.918691, auprc: 0.867024, aunpvspec: 0.965159, auprg: 0.947008, F1: 0.781198, recallAt0.05FDR: 0.395861, recallAt0.1FDR: 0.592790
# Performance on shared macaque cortex and caudate-putamen enhancers and macaque caudate-putamen enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1769
# Negatives: 1081
# Performance: loss: 0.460997, accuracy: 0.802807, sensitivity: 0.795365, specificity: 0.814986, auc: 0.885539, precision: 0.875544, NPV: 0.708769, auprc: 0.821645, aunpvspec: 0.926582, auprg: 0.839170, F1: 0.833531, recallAt0.05FDR: 0.133673, recallAt0.1FDR: 0.370490
# Performance on shared macaque cortex and liver enhancers and macaque liver enhancers that are not cortex enhancers 
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 657
# Negatives: 2624
# Performance: loss: 0.303064, accuracy: 0.871579, sensitivity: 0.863014, specificity: 0.876106, auc: 0.942481, precision: 0.786408, NPV: 0.923664, auprc: 0.894730, aunpvspec: 0.969170, auprg: 0.933518, F1: 0.822932, recallAt0.05FDR: 0.306697, recallAt0.1FDR: 0.657534
# Performance on shared rat cortex and striatum enhancers and rat striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 935
# Performance: loss: 0.688183, accuracy: 0.712439, sensitivity: 0.683279, specificity: 0.797861, auc: 0.814529, precision: 0.908275, NPV: 0.462349, auprc: 0.589246, aunpvspec: 0.925059, auprg: 0.754168, F1: 0.779873, recallAt0.05FDR: 0.013904, recallAt0.1FDR: 0.020321
# Performance on rat cortex and liver enhancers and rat liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1136
# Negatives: 1367
# Performance: loss: 0.434299, accuracy: 0.821614, sensitivity: 0.716989, specificity: 0.908559, auc: 0.903885, precision: 0.866951, NPV: 0.794372, auprc: 0.899012, aunpvspec: 0.905499, auprg: 0.864389, F1: 0.784871, recallAt0.05FDR: 0.536532, recallAt0.1FDR: 0.657570
# Performance on shared bat cortex and striatum enhancers and bat striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhPeakLessThan1kb500bp_data_train.KerasModels/cortexEnhancerShort_10XGCAndOtherTissueEnhNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolThreeDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 886
# Negatives: 746
# Performance: loss: 0.401665, accuracy: 0.829657, sensitivity: 0.838036, specificity: 0.819705, auc: 0.906124, precision: 0.846636, NPV: 0.809934, auprc: 0.890655, aunpvspec: 0.915534, auprg: 0.858508, F1: 0.842314, recallAt0.05FDR: 0.361260, recallAt0.1FDR: 0.641421