# Get positive test set for cortex
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz | grep -P 'chr1\t|chr2\t' | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Macaque
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Rat
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Bat
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat.fa

# Get positive test set for cortex, 2-species model
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque.fa

# Get positive test set for liver
# Macaque
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Rat
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Cow
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.narrowPeak.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/bosTau2015/GCF_000003205.7_Btau_5.0.1_genomic_withChrM.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Pig
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort.narrowPeak.gz --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/susScr3/susScr3.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_summitPlusMinus250bp_andMacaque_andRat_andCow_andPig_test.fa

# Get negative test set for cortex
# Mouse
gunzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed.gz
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | grep -P 'chr1\t|chr2\t' | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp.fa
# Macaque
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_notInMacaque.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_notInMacaque_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_notInMacaque_test.bed | grep -v chrUn | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa
# Rat
gunzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed.gz
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_test_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rn6_summitExtendedMin50Max2XProtect5_notInRat.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rn6_summitExtendedMin50Max2XProtect5_notInRat_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_test_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_rn6_summitExtendedMin50Max2XProtect5_notInRat_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa
# Bat
gunzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed.gz
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test.bed | awk 'BEGIN{OFS="\t"} $2+$10-250 >= 0' | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andBatNeg.fa

# Get negative test set for cortex, 2-species model
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMacaqueNeg.fa

# Get negative test set for cortex, 2-species, 10X G/C-repeat-matched and enhancer in other brain region negatives model
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_unique_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseCortex_andOtherNonMotorCortexBrainRegions.narrowPeak.gz | grep -P 'chr1\t|chr2\t' | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_unique_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseCortex_andOtherNonMotorCortexBrainRegions_test_summitPlusMinus250bp.fa
# Macaque
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_andStrCbNonCortex_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_andStrCbNonCortex_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_withPeakNames_andStrCbNonCortex.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_andStrCbNonCortex_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_andStrCbNonCortex_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_andStrCbNonCortex_test.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMacaqueCortex_andStrCbNonCortex.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMacaqueCortex_andStrCbNonCortex_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMacaqueCortex_andStrCbNonCortex_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMacaqueCortex_andStrCbNonCortex_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_unique_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseCortex_andOtherNonMotorCortexBrainRegions_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMacaqueCortex_andStrCbNonCortex_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_unique_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseCortex_andOtherNonMotorCortexBrainRegions_test_summitPlusMinus250bp_andMacaqueNeg.fa

# Get negative test set for liver
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp.fa
# Macaque
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test.bed
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test.bed
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver_test.bed
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComp18819utationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rheMac8Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonMacaqueLiver_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig.bed | awk 'BEGIN{OFS="\t"} $2 >= 0' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp.fa
# Rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_rn6Fixed_summitExtendedMin50Max2Protect5_nonRatLiver_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp.fa
# Cow
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/bosTau2015/GCF_000003205.7_Btau_5.0.1_genomic_withChrM.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp.fa
# Pig
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver_test.bed
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver_test.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow.bed | awk 'BEGIN{OFS="\t"} $2 >= 0' | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/susScr3/susScr3.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andCowNeg_andPigNeg.fa

# Get species-specific cortex enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats.bed.gz | grep -P 'chr1\t|chr2\t' >  /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp.fa
# Rat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseCortex_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa

# Get species-specific liver enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test_summitPlusMinus250bp.fa
# Rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_test_summitPlusMinus250bp.fa
# Cow
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonPigLiver_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonPigLiver_test.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/bosTau2015/GCF_000003205.7_Btau_5.0.1_genomic_withChrM.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonPigLiver_test_summitPlusMinus250bp.fa
# Pig
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonCowLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonCowLiver_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonCowLiver_test.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/susScr3/susScr3.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonCowLiver_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonPigLiver_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_nonCowLiver_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test_summitPlusMinus250bp_andRatSpecific_andCowSpecific_andPigSpecific.fa

# Get species-specific cortex non-enhancers
# Mouse
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp.fa
# Rat
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa

# Get species-specific liver non-enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test_summitPlusMinus250bp.fa
# Rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_summitPlusMinus250bp.fa
# Cow
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver_test.bed | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/bosTau2015/GCF_000003205.7_Btau_5.0.1_genomic_withChrM.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver_test_summitPlusMinus250bp.fa
# Pig
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver_test.bed | awk 'BEGIN{OFS="\t"} $2 >= 0' | fastaFromBed -fi /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/susScr3/susScr3.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_bosTau2015Fixed_summitExtendedMin50Max2Protect5_RefSeqNames_nonCowLiver_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_GenBankNames_susScr3Fixed_summitExtendedMin50Max2Protect5_UCSCNames_nonPigLiver_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test_summitPlusMinus250bp_andMouseNonRatLiver_andPigNonCowLiver_andCowNonPigLiver.fa

# Get non-restrictive euarchonta-specific liver enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_nonRatLiverFixed_nonCowLiver_nonPigLiver.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_nonRatLiverFixed_nonCowLiver_nonPigLiver_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_nonRatLiverFixed_nonCowLiver_nonPigLiver_test.bed

# Get non-restrictive euarchonta-specific liver non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_inRatLiverLoose_inCowLiverLoose_inPigLiverLoose.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_inRatLiverLoose_inCowLiverLoose_inPigLiverLoose_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_inRatLiverLoose_inCowLiverLoose_inPigLiverLoose_test.bed

# Get shared cortex and striatum enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp.fa
# Macaque
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test_summitPlusMinus250bp.fa
# Rat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test_summitPlusMinus250bp.fa
# Bat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inCaudatePutamen_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inStriatum_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_inStriatum_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa

# Get striatum enhancers that are not cortex enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp.fa
# Macaque
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa
# Rat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa
# Bat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/StrP_AllReps/peak/idr_reproducibility/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Striatum/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/StrPBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inStrC_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa

# Get shared cortex and liver enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats.bed.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp.fa
# Macaque
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test_summitPlusMinus250bp.fa
# Rat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_inLiver_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inLiver_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa

# Get shared liver and cortex enhancers
# Mouse
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test.bed.gz -b /projects/pfenninggroup/machineLearningForComputationalBiology/addiction_gwas_enrichment/mouse_enrichment/peaks/Pfenning_bulk_Ctx.narrowPeak.gz > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa
# Macaque
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_inCortex.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_inCortex.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa
# Rat
intersectBed -wa -u -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_inCortex.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_inCortex.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp_andMacaqueLiverCortex_andRatLiverCortex.fa

# Get liver enhancers that are not cortex enhancers
# Mouse
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex.narrowPeak.gz | grep -P 'chr1\t|chr2\t' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp.fa
# Macaque
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa
# Rat
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test.bed.gz | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_notInCortex_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa

# Get cortex enhancers that are not liver enhancers
# Mouse
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-call_peak_pooled/execution/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L2_1.trim.merged.nodup.no_chrM.tn5.pooled.pval0.01.300K.bfilt.narrowPeak.gz | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverAllLiu_ATAC_out/atac/ac63679d-3e9b-45e7-8fa5-b7b221cb6aaa/peak/pooled-rep/Male_Liver_Rep1_1.merged.nodup.tn5.pooled.pval0.01.300K.bfilt.narrowPeak.gz > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver_summitPlusMinus250bp.fa
# Macaque
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test.bed -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-macs2_pooled/execution/Mac1_Lv_1_PFE116A29_S8_R1_001.merged.nodup.tn5.pooled.pval0.01.300K.bfilt.narrowPeak.gz > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_nonLiver.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_nonLiver.bed | awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_nonLiver_summitPlusMinus250bp.fa
# Rat
subtractBed -A -a /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test.bed -b /projects/MPRA/Irene/rats/atac-pipeline-output/Liver/peak/pooled-rep/R2-Liv_R1_001.trim.merged.nodup.no_chrM.tn5.pooled.pval0.01.300K.bfilt.narrowPeak.gz > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_nonLiver.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10-250, $2+$10+250, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_nonLiver.bed | fastaFromBed -fi /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatGenome/rn6.fa -bed stdin -fo /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_nonLiver_summitPlusMinus250bp.fa
# Combine
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_nonLiver_summitPlusMinus250bp.fa /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_nonLiver_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver_summitPlusMinus250bp_andMacaqueCortexNonLiver_andRatCortexNonLiver.fa

# Get Gliobla enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk.narrowPeak.gz | awk '{printf("%s\tpeak%d\n", $0, NR)}' | awk 'BEGIN{OFS="\t"} {print $1,$2,$3,$11,$5,$6,$7,$8,$9,$10}' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$7, $2+$7+1, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_summits.bed
awk 'BEGIN{OFS="\t"} {print $1, $2, $3, $4, $5, $6, int($7), $8, $9, $10}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames.bed | /home/ikaplow/RegulatoryElementEvolutionProject/src/liftOver stdin /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/LiftoverChains/hg19ToHg38.over.chain.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_unmapped.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/liftOver /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_summits.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/LiftoverChains/hg19ToHg38.over.chain.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_summits_hg38.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_summits_unmapped.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/halLiftover-postprocessing/orthologFind.py -max_frac 2 -min_len 50 -protect_dist 5 -qFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames.bed -tFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38.bed -sFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_summits_hg38.bed -oFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5.bed -narrowPeak
awk 'BEGIN{OFS="\t"} $3-$2 <= 1000' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5.bed | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protCDS_geneNames.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | closestBed -d -t first -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protTranscript.TSSsWithStrand_sorted.bed | awk 'BEGIN{OFS="\t"} $17 > 20000' | cut -f1-10 > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10, $2+$10+1, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_summits.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/hal/bin/halLiftover --bedType 4 /data/pfenninggroup/machineLearningForComputationalBiology/alignCactus/mam241/241-mammalian-2020v2.hal Homo_sapiens /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed Mus_musculus /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/hal/bin/halLiftover --bedType 4 /data/pfenninggroup/machineLearningForComputationalBiology/alignCactus/mam241/241-mammalian-2020v2.hal Homo_sapiens /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_summits.bed Mus_musculus /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_summits_mm10.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/halLiftover-postprocessing/orthologFind.py -max_frac 2 -min_len 50 -protect_dist 5 -qFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed -tFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10.bed -sFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_summits_mm10.bed -oFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed -narrowPeak
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed

# Get reprocessed Gliobla enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/atac/081b7352-d06c-4077-a623-b11cc996306a/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak.gz | awk 'BEGIN{OFS="\t"} $3-$2 <= 1000' | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protCDS_geneNames.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | closestBed -d -t first -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protTranscript.TSSsWithStrand_sorted.bed | awk 'BEGIN{OFS="\t"} $17 > 20000' | cut -f1-10 | awk '{printf("%s\tpeak%d\n", $0, NR)}' | awk 'BEGIN{OFS="\t"} {print $1,$2,$3,$11,$5,$6,$7,$8,$9,$10}' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/atac/081b7352-d06c-4077-a623-b11cc996306a/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/atac/081b7352-d06c-4077-a623-b11cc996306a/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed

# Get Hapatocyte enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk.narrowPeak.gz | awk '{printf("%s\tpeak%d\n", $0, NR)}' | awk 'BEGIN{OFS="\t"} {print $1,$2,$3,$11,$5,$6,$7,$8,$9,$10}' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$7, $2+$7+1, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_summits.bed
awk 'BEGIN{OFS="\t"} {print $1, $2, $3, $4, $5, $6, int($7), $8, $9, $10}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames.bed | /home/ikaplow/RegulatoryElementEvolutionProject/src/liftOver stdin /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/LiftoverChains/hg19ToHg38.over.chain.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_unmapped.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/liftOver /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_summits.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/LiftoverChains/hg19ToHg38.over.chain.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_summits_hg38.bed /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_summits_unmapped.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/halLiftover-postprocessing/orthologFind.py -max_frac 2 -min_len 50 -protect_dist 5 -qFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames.bed -tFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38.bed -sFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_summits_hg38.bed -oFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5.bed -narrowPeak
awk 'BEGIN{OFS="\t"} $3-$2 <= 1000' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5.bed | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protCDS_geneNames.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | closestBed -d -t first -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protTranscript.TSSsWithStrand_sorted.bed | awk 'BEGIN{OFS="\t"} $17 > 20000' | cut -f1-10 > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed
awk 'BEGIN{OFS="\t"} {print $1, $2+$10, $2+$10+1, $4}' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_summits.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/hal/bin/halLiftover --bedType 4 /data/pfenninggroup/machineLearningForComputationalBiology/alignCactus/mam241/241-mammalian-2020v2.hal Homo_sapiens /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed Mus_musculus /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10.bed
/home/ikaplow/RegulatoryElementEvolutionProject/src/hal/bin/halLiftover --bedType 4 /data/pfenninggroup/machineLearningForComputationalBiology/alignCactus/mam241/241-mammalian-2020v2.hal Homo_sapiens /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_summits.bed Mus_musculus /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_summits_mm10.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/halLiftover-postprocessing/orthologFind.py -max_frac 2 -min_len 50 -protect_dist 5 -qFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed -tFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10.bed -sFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_summits_mm10.bed -oFile /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed -narrowPeak
grep -P 'chr1\t|chr2\t' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5.bed > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed 
python /home/ikaplow/RegulatoryElementEvolutionProject/src/OCROrthologPrediction/src/filterPeakName.py --unfilteredPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort.bed --peakListFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_test.bed --peakNameCol 3 --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed

# Get reprocessed Hepatocyte enhancers
zcat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/atac/cb929f98-9900-4455-ad6a-48853584c300/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak.gz | awk 'BEGIN{OFS="\t"} $3-$2 <= 1000' | subtractBed -A -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protCDS_geneNames.bed | sort -u -k1,1 -k2,2n -k3,3n -k10,10n | closestBed -d -t first -a stdin -b /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/gencode.v27.annotation.protTranscript.TSSsWithStrand_sorted.bed | awk 'BEGIN{OFS="\t"} $17 > 20000' | cut -f1-10 | awk '{printf("%s\tpeak%d\n", $0, NR)}' | awk 'BEGIN{OFS="\t"} {print $1,$2,$3,$11,$5,$6,$7,$8,$9,$10}' > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/atac/cb929f98-9900-4455-ad6a-48853584c300/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/atac/cb929f98-9900-4455-ad6a-48853584c300/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed

# Evaluate motor cortex model performance on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andBatNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat_andNeg_predictedProba.txt
# Positives: 20820
# Negatives: 21613
# Performance: loss: 0.384362, accuracy: 0.827045, sensitivity: 0.845245, specificity: 0.809513, auc: 0.908534, precision: 0.810408, NPV: 0.844483, auprc: 0.901379, aunpvspec: 0.915066, F1: 0.827460, recallAt0.05FDR: 0.395653, recallAt0.1FDR: 0.639433
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4843
# Negatives: 4843
# Performance: loss: 0.496821, accuracy: 0.760789, sensitivity: 0.837084, specificity: 0.684493, auc: 0.848658, precision: 0.726263, NPV: 0.807749, auprc: 0.842140, aunpvspec: 0.850448, auprg: 0.713795, F1: 0.777746, recallAt0.05FDR: 0.174582, recallAt0.1FDR: 0.392319
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4153
# Negatives: 1874
# Performance: loss: 0.337850, accuracy: 0.857461, sensitivity: 0.845295, specificity: 0.885609, auc: 0.935442, precision: 0.944742, NPV: 0.712166, auprc: 0.878459, aunpvspec: 0.967784, auprg: 0.944181, F1: 0.892256, recallAt0.05FDR: 0.409594, recallAt0.1FDR: 0.640221
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.405563, accuracy: 0.820232, sensitivity: 0.850694, specificity: 0.732500, auc: 0.872285, precision: 0.901564, NPV: 0.630108, auprc: 0.760258, aunpvspec: 0.933900, F1: 0.875391, recallAt0.05FDR: 0.212500, recallAt0.1FDR: 0.327500
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.370836, accuracy: 0.835402, sensitivity: 0.851205, specificity: 0.818623, auc: 0.914998, precision: 0.832859, NPV: 0.838229, auprc: 0.920465, aunpvspec: 0.904813, auprg: 0.871015, F1: 0.841932, recallAt0.05FDR: 0.612533, recallAt0.1FDR: 0.733010
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.358496, accuracy: 0.849874, sensitivity: 0.875439, specificity: 0.835616, auc: 0.922831, precision: 0.748126, NPV: 0.923243, auprc: 0.848603, aunpvspec: 0.959772, auprg: 0.886507, F1: 0.806791, recallAt0.05FDR: 0.092982, recallAt0.1FDR: 0.368421
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa --sequenceLength 500 --RC --classification --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum_andNeg_predictedProba.txt
# Positives: 8461
# Negatives: 6523
# Performance: loss: 0.496688, accuracy: 0.762380, sensitivity: 0.848186, specificity: 0.651081, auc: 0.844142, precision: 0.759217, NPV: 0.767785, auprc: 0.804923, aunpvspec: 0.872848, auprg: 0.738157, F1: 0.801239, recallAt0.05FDR: 0.123256, recallAt0.1FDR: 0.269048
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 2477
# Negatives: 4578
# Performance: loss: 0.335258, accuracy: 0.860524, sensitivity: 0.849415, specificity: 0.866536, auc: 0.930891, precision: 0.774954, NPV: 0.914055, auprc: 0.898375, aunpvspec: 0.956803, auprg: 0.934083, F1: 0.810478, recallAt0.05FDR: 0.547033, recallAt0.1FDR: 0.683488
# Performance on mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.378193, accuracy: 0.829822, sensitivity: 0.849975, specificity: 0.798882, auc: 0.907867, precision: 0.866462, NPV: 0.776206, auprc: 0.866975, aunpvspec: 0.937277, auprg: 0.873248, F1: 0.858139, recallAt0.05FDR: 0.297821, recallAt0.1FDR: 0.513787
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 5807
# Performance: loss: 0.477505, accuracy: 0.771447, sensitivity: 0.811009, specificity: 0.723179, auc: 0.852155, precision: 0.781397, NPV: 0.758238, auprc: 0.830909, aunpvspec: 0.870113, auprg: 0.752999, F1: 0.795928, recallAt0.05FDR: 0.210350, recallAt0.1FDR: 0.388238
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3762
# Negatives: 5733
# Performance: loss: 0.331854, accuracy: 0.859874, sensitivity: 0.878921, specificity: 0.847375, auc: 0.936513, precision: 0.790745, NPV: 0.914275, auprc: 0.902126, aunpvspec: 0.959080, auprg: 0.914811, F1: 0.832505, recallAt0.05FDR: 0.410553, recallAt0.1FDR: 0.655369
# Performance on bat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2011
# Negatives: 4887
# Performance: loss: 0.294316, accuracy: 0.880473, sensitivity: 0.884137, specificity: 0.878965, auc: 0.950450, precision: 0.750369, NPV: 0.948548, auprc: 0.880599, aunpvspec: 0.979337, auprg: 0.948420, F1: 0.811779, recallAt0.05FDR: 0.270512, recallAt0.1FDR: 0.588513
# Performance on Gliobla enhancers
# Positives: 4044
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/wgEncodeAwgDnaseDukeGlioblaUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed.gz --label 1 --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/hg38.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning # Accuracy: 0.265702
# Performance on reprocessed Gliobla enhancers
# Positives: 1602
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5TotalTotalFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Gliobla/atac/081b7352-d06c-4077-a623-b11cc996306a/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed.gz --label 1 --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/hg38.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning # Accuracy: 0.490225

# Evaluate inferior motor cortex model performance on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andBatNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 20820
# Negatives: 21613
# Performance: loss: nan, accuracy: 0.802206, sensitivity: 0.734606, specificity: 0.867325, auc: 0.891014, precision: 0.842115, NPV: 0.772341, auprc: 0.883800, aunpvspec: 0.897421, F1: 0.784696, recallAt0.05FDR: 0.318804, recallAt0.1FDR: 0.563521
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4843
# Negatives: 4843
# Performance: loss: nan, accuracy: 0.746077, sensitivity: 0.723106, specificity: 0.769048, auc: 0.823070, precision: 0.757927, NPV: 0.735268, auprc: 0.818373, aunpvspec: 0.820863, auprg: 0.663314, F1: 0.740107, recallAt0.05FDR: 0.101177, recallAt0.1FDR: 0.291865
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4153
# Negatives: 1874
# Performance: loss: 0.580701, accuracy: 0.795657, sensitivity: 0.744817, specificity: 0.913284, auc: 0.904111, precision: 0.952090, NPV: 0.607362, auprc: 0.806998, aunpvspec: 0.952739, auprg: 0.889469, F1: 0.835794, recallAt0.05FDR: 0.129151, recallAt0.1FDR: 0.311808
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.749063, accuracy: 0.717784, sensitivity: 0.685764, specificity: 0.810000, auc: 0.836263, precision: 0.912240, NPV: 0.472303, auprc: 0.697734, aunpvspec: 0.918911, F1: 0.782953, recallAt0.05FDR: 0.112500, recallAt0.1FDR: 0.220000
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.552885, accuracy: 0.783604, sensitivity: 0.698670, specificity: 0.873786, auc: 0.889198, precision: 0.854601, NPV: 0.731978, auprc: 0.893515, aunpvspec: 0.879195, auprg: 0.820567, F1: 0.768809, recallAt0.05FDR: 0.474846, recallAt0.1FDR: 0.644307
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.424256, accuracy: 0.819724, sensitivity: 0.710526, specificity: 0.880626, auc: 0.903078, precision: 0.768501, NPV: 0.845070, auprc: 0.829307, aunpvspec: 0.946755, auprg: 0.867564, F1: 0.738377, recallAt0.05FDR: 0.133333, recallAt0.1FDR: 0.389474
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 8461
# Negatives: 6523
# Performance: loss: nan, accuracy: 0.748231, sensitivity: 0.745952, specificity: 0.751188, auc: 0.830136, precision: 0.795450, NPV: 0.695085, auprc: 0.788805, aunpvspec: 0.861758, auprg: 0.706027, F1: 0.769906, recallAt0.05FDR: 0.123180, recallAt0.1FDR: 0.245593
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 2477
# Negatives: 4578
# Performance: loss: nan, accuracy: 0.853792, sensitivity: 0.759386, specificity: 0.904871, auc: 0.916155, precision: 0.812001, NPV: 0.874222, auprc: 0.870880, aunpvspec: 0.948337, auprg: 0.912230, F1: 0.784813, recallAt0.05FDR: 0.407348, recallAt0.1FDR: 0.611021
# Performance on mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.545869, accuracy: 0.790729, sensitivity: 0.747614, specificity: 0.856922, auc: 0.883779, precision: 0.889163, NPV: 0.688619, auprc: 0.832710, aunpvspec: 0.919751, auprg: 0.829710, F1: 0.812268, recallAt0.05FDR: 0.222233, recallAt0.1FDR: 0.379772
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 5807
# Performance: loss: nan, accuracy: 0.750194, sensitivity: 0.711503, specificity: 0.797400, auc: 0.836972, precision: 0.810776, NPV: 0.693760, auprc: 0.809810, aunpvspec: 0.858751, auprg: 0.718449, F1: 0.757903, recallAt0.05FDR: 0.144997, recallAt0.1FDR: 0.296194
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3762
# Negatives: 5733
# Performance: loss: 0.414236, accuracy: 0.835387, sensitivity: 0.738969, specificity: 0.898657, auc: 0.915920, precision: 0.827135, NPV: 0.839909, auprc: 0.875300, aunpvspec: 0.943743, auprg: 0.885451, F1: 0.780570, recallAt0.05FDR: 0.318581, recallAt0.1FDR: 0.556619
# Performance on bat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5MuchMuchMoreFiltNoInterPoolTwoDenseLargeDenseVeryHighMomLowDropL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2011
# Negatives: 4887
# Performance: loss: nan, accuracy: 0.875616, sensitivity: 0.756340, specificity: 0.924698, auc: 0.937015, precision: 0.805188, NPV: 0.902176, auprc: 0.856652, aunpvspec: 0.972952, auprg: 0.935646, F1: 0.780000, recallAt0.05FDR: 0.225012, recallAt0.1FDR: 0.472899

# Evaluate substantially inferior motor cortex model performance on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andBatNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 20820
# Negatives: 21613
# Performance: loss: 0.489177, accuracy: 0.763781, sensitivity: 0.901465, specificity: 0.631148, auc: 0.869997, precision: 0.701875, NPV: 0.869269, auprc: 0.857726, aunpvspec: 0.874216, F1: 0.789247, recallAt0.05FDR: 0.143348, recallAt0.1FDR: 0.465802
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4843
# Negatives: 4843
# Performance: loss: 0.603092, accuracy: 0.677679, sensitivity: 0.906773, specificity: 0.448586, auc: 0.795024, precision: 0.621849, NPV: 0.827934, auprc: 0.778666, aunpvspec: 0.789544, F1: 0.737757, recallAt0.05FDR: 0.001136, recallAt0.1FDR: 0.048833
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4153
# Negatives: 1874
# Performance: loss: 0.420501, accuracy: 0.825724, sensitivity: 0.912281, specificity: 0.625461, auc: 0.874954, precision: 0.849295, NPV: 0.755011, auprc: 0.775334, aunpvspec: 0.927772, auprg: 0.860705, F1: 0.879662, recallAt0.05FDR: 0.145756, recallAt0.1FDR: 0.287823
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.446378, accuracy: 0.828608, sensitivity: 0.910590, specificity: 0.592500, auc: 0.831042, precision: 0.865512, NPV: 0.697059, auprc: 0.691208, aunpvspec: 0.903794, F1: 0.887479, recallAt0.05FDR: 0.075000, recallAt0.1FDR: 0.140000
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.467960, accuracy: 0.802226, sensitivity: 0.908978, specificity: 0.688879, auc: 0.880464, precision: 0.756224, NPV: 0.876966, auprc: 0.884625, aunpvspec: 0.854477, auprg: 0.811072, F1: 0.825595, recallAt0.05FDR: 0.396293, recallAt0.1FDR: 0.656222
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.493657, accuracy: 0.766960, sensitivity: 0.942105, specificity: 0.669276, auc: 0.896708, precision: 0.613714, NPV: 0.953975, auprc: 0.796276, aunpvspec: 0.942347, auprg: 0.854552, F1: 0.743253, recallAt0.05FDR: 0.008772, recallAt0.1FDR: 0.268421
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 8461
# Negatives: 6523
# Performance: loss: 0.580638, accuracy: 0.676121, sensitivity: 0.887957, specificity: 0.401349, auc: 0.793342, precision: 0.657996, NPV: 0.734156, auprc: 0.703991, aunpvspec: 0.848406, auprg: 0.622248, F1: 0.755873, recallAt0.05FDR: 0.000383, recallAt0.1FDR: 0.000383
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 2477
# Negatives: 4578
# Performance: loss: 0.434837, accuracy: 0.795393, sensitivity: 0.889584, specificity: 0.744430, auc: 0.913670, precision: 0.653179, NPV: 0.925710, auprc: 0.871999, aunpvspec: 0.945065, auprg: 0.913943, F1: 0.753269, recallAt0.05FDR: 0.461042, recallAt0.1FDR: 0.643117
# Performance on mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.463479, accuracy: 0.784340, sensitivity: 0.912396, specificity: 0.587736, auc: 0.862838, precision: 0.772614, NPV: 0.813777, auprc: 0.810725, aunpvspec: 0.896086, auprg: 0.803041, F1: 0.836707, recallAt0.05FDR: 0.132279, recallAt0.1FDR: 0.318550
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 5807
# Performance: loss: 0.554257, accuracy: 0.711061, sensitivity: 0.872759, specificity: 0.513776, auc: 0.805075, precision: 0.686522, NPV: 0.767954, auprc: 0.762920, aunpvspec: 0.829677, auprg: 0.648348, F1: 0.768519, recallAt0.05FDR: 0.042621, recallAt0.1FDR: 0.157052
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3762
# Negatives: 5733
# Performance: loss: 0.464414, accuracy: 0.787204, sensitivity: 0.914274, specificity: 0.703820, auc: 0.903620, precision: 0.669489, NPV: 0.925990, auprc: 0.853516, aunpvspec: 0.933501, auprg: 0.864953, F1: 0.772965, recallAt0.05FDR: 0.120415, recallAt0.1FDR: 0.463583
# Performance on bat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaqueRatBatPeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_multiSpecies_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolNoDenseVeryHighMomLowLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2011
# Negatives: 4887
# Performance: loss: 0.450616, accuracy: 0.790881, sensitivity: 0.935356, specificity: 0.731430, auc: 0.925066, precision: 0.589009, NPV: 0.964908, auprc: 0.841793, aunpvspec: 0.966530, auprg: 0.923881, F1: 0.722836, recallAt0.05FDR: 0.215067, recallAt0.1FDR: 0.446295

# Evaluate 2-species motor cortex model on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMacaqueNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 11724
# Negatives: 8125
# Performance: loss: 0.396256, accuracy: 0.822586, sensitivity: 0.912700, specificity: 0.692554, auc: 0.901562, precision: 0.810736, NPV: 0.846102, auprc: 0.875258, aunpvspec: 0.923368, auprg: 0.868563, F1: 0.858702, recallAt0.05FDR: 0.386585, recallAt0.1FDR: 0.579508
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4843
# Negatives: 4843
# Performance: loss: 0.634455, accuracy: 0.691462, sensitivity: 0.891699, specificity: 0.491224, auc: 0.806367, precision: 0.636712, NPV: 0.819356, auprc: 0.799073, aunpvspec: 0.802370, auprg: 0.627114, F1: 0.742936, recallAt0.05FDR: 0.055854, recallAt0.1FDR: 0.211336
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4153
# Negatives: 1874
# Performance: loss: 0.364648, accuracy: 0.839644, sensitivity: 0.905901, specificity: 0.686347, auc: 0.896688, precision: 0.869832, NPV: 0.759184, auprc: 0.808346, aunpvspec: 0.946675, auprg: 0.891697, F1: 0.887500, recallAt0.05FDR: 0.158672, recallAt0.1FDR: 0.315498
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.403137, accuracy: 0.833763, sensitivity: 0.921875, specificity: 0.580000, auc: 0.843791, precision: 0.863415, NPV: 0.720497, auprc: 0.731014, aunpvspec: 0.920419, F1: 0.891688, recallAt0.05FDR: 0.212500, recallAt0.1FDR: 0.305000
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.430621, accuracy: 0.811216, sensitivity: 0.923940, specificity: 0.691527, auc: 0.899680, precision: 0.760780, NPV: 0.895429, auprc: 0.906283, aunpvspec: 0.885855, auprg: 0.844149, F1: 0.834459, recallAt0.05FDR: 0.541924, recallAt0.1FDR: 0.685790
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.571472, accuracy: 0.733040, sensitivity: 0.917544, specificity: 0.630137, auc: 0.870733, precision: 0.580466, NPV: 0.931983, auprc: 0.750502, aunpvspec: 0.929895, auprg: 0.800333, F1: 0.711081, recallAt0.05FDR: 0.000000, recallAt0.1FDR: 0.085965
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa --sequenceLength 500 --RC --classification --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum_andNeg_predictedProba.txt
# Positives: 8461
# Negatives: 6523
# Performance: loss: 0.607228, accuracy: 0.704485, sensitivity: 0.886420, specificity: 0.468496, auc: 0.803989, precision: 0.683870, NPV: 0.760767, auprc: 0.746984, aunpvspec: 0.842284, auprg: 0.649357, F1: 0.772082, recallAt0.05FDR: 0.037253, recallAt0.1FDR: 0.092825
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 2477
# Negatives: 4578
# Performance: loss: 0.438755, accuracy: 0.799717, sensitivity: 0.891401, specificity: 0.750109, auc: 0.915347, precision: 0.658711, NPV: 0.927356, auprc: 0.867648, aunpvspec: 0.947289, auprg: 0.910058, F1: 0.757591, recallAt0.05FDR: 0.350020, recallAt0.1FDR: 0.598304
# Performance on mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.436750, accuracy: 0.795292, sensitivity: 0.905112, specificity: 0.626687, auc: 0.878346, precision: 0.788242, NPV: 0.811384, auprc: 0.830895, aunpvspec: 0.912782, auprg: 0.825993, F1: 0.842644, recallAt0.05FDR: 0.212688, recallAt0.1FDR: 0.388835
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 5807
# Performance: loss: 0.575289, accuracy: 0.722813, sensitivity: 0.864291, specificity: 0.550198, auc: 0.809246, precision: 0.700990, NPV: 0.768676, auprc: 0.777291, aunpvspec: 0.832768, auprg: 0.659089, F1: 0.774122, recallAt0.05FDR: 0.100396, recallAt0.1FDR: 0.203978
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3762
# Negatives: 5733
# Performance: loss: 0.433134, accuracy: 0.804160, sensitivity: 0.928761, specificity: 0.722397, auc: 0.919854, precision: 0.687051, NPV: 0.939222, auprc: 0.876358, aunpvspec: 0.947455, auprg: 0.887200, F1: 0.789828, recallAt0.05FDR: 0.301568, recallAt0.1FDR: 0.533759
# Performance on bat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVsOtherSpeciesEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_enhLooseNeg_500bp_conv5AllAllFiltNoInterPoolLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2011
# Negatives: 4887
# Performance: loss: 0.469794, accuracy: 0.783343, sensitivity: 0.935356, specificity: 0.720790, auc: 0.919800, precision: 0.579572, NPV: 0.964408, auprc: 0.814782, aunpvspec: 0.965849, auprg: 0.907074, F1: 0.715685, recallAt0.05FDR: 0.134262, recallAt0.1FDR: 0.307558

# Evaluate 2-species, 10X G/C-repeat-matched and enhancer in other brain region negative model performance on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_unique_summitPlusMinus250bp_randomGCRepeatMatched10x_nonCDS_nonPromoter_nonMouseCortex_andOtherNonMotorCortexBrainRegions_test_summitPlusMinus250bp_andMacaqueNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 11724
# Negatives: 46801
# Performance: loss: 0.198183, accuracy: 0.919590, sensitivity: 0.819089, specificity: 0.944766, auc: 0.962704, precision: 0.787906, NPV: 0.954227, auprc: 0.882674, aunpvspec: 0.989944, auprg: 0.980012, F1: 0.803195, recallAt0.05FDR: 0.454495, recallAt0.1FDR: 0.637027
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_nonRatCortex_noRepeats_test_summitPlusMinus250bp_andRatSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_summitPlusMinus250bp_andMouseNonRatCortex.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4843
# Negatives: 4843
# Performance: loss: 0.559111, accuracy: 0.734823, sensitivity: 0.763267, specificity: 0.706380, auc: 0.815056, precision: 0.722184, NPV: 0.748987, auprc: 0.800429, aunpvspec: 0.815883, auprg: 0.645281, F1: 0.742157, recallAt0.05FDR: 0.002375, recallAt0.1FDR: 0.166632
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inRatLoose_notInMacaque_notInBat_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_notInRat_inBatLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 4153
# Negatives: 1874
# Performance: loss: 0.409402, accuracy: 0.816815, sensitivity: 0.803030, specificity: 0.848708, auc: 0.906453, precision: 0.924702, NPV: 0.650636, auprc: 0.834201, aunpvspec: 0.951549, auprg: 0.913047, F1: 0.859582, recallAt0.05FDR: 0.247232, recallAt0.1FDR: 0.520295
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_notInBat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_inBatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 576
# Negatives: 200
# Performance: loss: 0.472209, accuracy: 0.794459, sensitivity: 0.828125, specificity: 0.697500, auc: 0.829125, precision: 0.887442, NPV: 0.584906, auprc: 0.702725, aunpvspec: 0.910315, F1: 0.856758, recallAt0.05FDR: 0.160000, recallAt0.1FDR: 0.282500
# Performance on non-restrictive euarchonta-specific relative to glires enhancers and non-restrictive euarchonta-specific relative to glires non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1203
# Negatives: 1133
# Performance: loss: 0.416029, accuracy: 0.812714, sensitivity: 0.825021, specificity: 0.799647, auc: 0.892540, precision: 0.813858, NPV: 0.811464, auprc: 0.900006, aunpvspec: 0.879450, auprg: 0.825589, F1: 0.819401, recallAt0.05FDR: 0.521183, recallAt0.1FDR: 0.639453
# Performance on bat-specific enhancers and non-restrictive bat-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_nonMouseCortex_notInMacaque_notInRat_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_orRatCortex_nonBatCortex_inMacaqueLoose_inRatLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatGenome/GCF_014176215.1_mRouAeg1.p_genomic_withMT.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 285
# Negatives: 511
# Performance: loss: 0.408223, accuracy: 0.820352, sensitivity: 0.878947, specificity: 0.787671, auc: 0.905304, precision: 0.697772, NPV: 0.921053, auprc: 0.805656, aunpvspec: 0.949866, auprg: 0.863353, F1: 0.777950, recallAt0.05FDR: 0.007018, recallAt0.1FDR: 0.298246
# Performance on shared cortex and striatum enhancers and striatum enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_inStriatum_test_summitPlusMinus250bp_andMacaqueCortexStriatum_andRatCortexStriatum_andBatCortexStriatum.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Striatum_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Str_nonCDS_enhancerShort_fixed_nonCortex_test_summitPlusMinus250bp_andMacaqueStriatumNonCortex_andRatStriatumNonCortex_andBatStriatumNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 8461
# Negatives: 6523
# Performance: loss: 0.434337, accuracy: 0.805459, sensitivity: 0.775795, specificity: 0.843937, auc: 0.893507, precision: 0.865735, NPV: 0.743718, auprc: 0.863463, aunpvspec: 0.916791, auprg: 0.832022, F1: 0.818301, recallAt0.05FDR: 0.271501, recallAt0.1FDR: 0.489805
# Performance on shared cortex and liver enhancers and liver enhancers that are not cortex enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_inLiver_noRepeats_test_summitPlusMinus250bp_andMacaqueCortexLiver_andRatCortexLiver.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_nonCortex_test_summitPlusMinus250bp_andMacaqueLiverNonCortex_andRatLiverNonCortex.fa --sequenceLength 500 --RC --classification
# Positives: 2477
# Negatives: 4578
# Performance: loss: 0.329143, accuracy: 0.863076, sensitivity: 0.799758, specificity: 0.897335, auc: 0.923703, precision: 0.808242, NPV: 0.892268, auprc: 0.889078, aunpvspec: 0.950549, auprg: 0.928020, F1: 0.803977, recallAt0.05FDR: 0.530682, recallAt0.1FDR: 0.683690
# Performance on mouse enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_andRat_andBat.narrowPeak.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7962
# Negatives: 5186
# Performance: loss: 0.434734, accuracy: 0.806168, sensitivity: 0.802876, specificity: 0.811223, auc: 0.888494, precision: 0.867191, NPV: 0.728296, auprc: 0.843129, aunpvspec: 0.920624, auprg: 0.845525, F1: 0.833795, recallAt0.05FDR: 0.227728, recallAt0.1FDR: 0.454396
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 7085
# Negatives: 5807
# Performance: loss: 0.569786, accuracy: 0.736658, sensitivity: 0.731475, specificity: 0.742983, auc: 0.816529, precision: 0.776404, NPV: 0.693984, auprc: 0.791457, aunpvspec: 0.833189, auprg: 0.678828, F1: 0.753270, recallAt0.05FDR: 0.129757, recallAt0.1FDR: 0.270019
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3762
# Negatives: 5733
# Performance: loss: 0.363513, accuracy: 0.836177, sensitivity: 0.853402, specificity: 0.824874, auc: 0.919629, precision: 0.761775, NPV: 0.895559, auprc: 0.876811, aunpvspec: 0.947929, auprg: 0.887956, F1: 0.804990, recallAt0.05FDR: 0.327485, recallAt0.1FDR: 0.538809
# Performance on bat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/cortexEnhancerVs10XGCAndOtherTissueEnhMouseMacaquePeakLessThan1kb_data_train.KerasModels/cortexEnhancerShort_2Species_10XGCAndOtherTissueEnhNeg_500bp_conv5AllAllFiltNoInterPool2DenseLargeDenseVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2011
# Negatives: 4887
# Performance: loss: 0.331616, accuracy: 0.852783, sensitivity: 0.880159, specificity: 0.841518, auc: 0.934860, precision: 0.695618, NPV: 0.944642, auprc: 0.850552, aunpvspec: 0.972601, auprg: 0.930940, F1: 0.777083, recallAt0.05FDR: 0.180259, recallAt0.1FDR: 0.456241

# Evaluate liver model performance on test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_summitPlusMinus250bp_andMacaque_andRat_andCow_andPig_test.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andCowNeg_andPigNeg.fa --predictedProbaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_summitPlusMinus250bp_andMacaque_andRat_andCow_andPig_test_andNeg_predictedProba.txt --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 12267
# Negatives: 18358
# Performance: loss: 0.418250, accuracy: 0.810563, sensitivity: 0.745048, specificity: 0.854341, auc: 0.884308, precision: 0.773649, NPV: 0.833745, auprc: 0.834676, aunpvspec: 0.915824, auprg: 0.833083, F1: 0.759079, recallAt0.05FDR: 0.133040, recallAt0.1FDR: 0.393332
# Performance on species-specific enhancers and non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_nonRatLiverFixed_test_summitPlusMinus250bp_andRatSpecific_andCowSpecific_andPigSpecific.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2XProtect5_nonMouseLiver_test_summitPlusMinus250bp_andMouseNonRatLiver_andPigNonCowLiver_andCowNonPigLiver.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3194
# Negatives: 3193
# Performance: loss: 0.529106, accuracy: 0.743620, sensitivity: 0.726362, specificity: 0.760883, auc: 0.818934, precision: 0.752392, NPV: 0.735432, auprc: 0.812046, aunpvspec: 0.813744, auprg: 0.655756, F1: 0.739148, recallAt0.05FDR: 0.090824, recallAt0.1FDR: 0.277012
# Performance on non-restrictive glires-specific enhancers and non-restrictive glires-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_inRatLiverLoose_nonMacaqueLiver_nonCowLiver_nonPigLiver.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_nonRatLiver_inCowLiverLoose_inPigLiverLoose.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 176
# Negatives: 69
# Performance: loss: 0.554834, accuracy: 0.728571, sensitivity: 0.687500, specificity: 0.833333, auc: 0.857790, precision: 0.913208, NPV: 0.511111, auprc: 0.741417, aunpvspec: 0.935833, auprg: 0.861233, F1: 0.784441, recallAt0.05FDR: 0.159420, recallAt0.1FDR: 0.376812
# Performance on non-restrictive euarchonta-specific enhancers and non-restrictive euarchonta-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_nonMouseLiverFixed_nonRatLiverFixed_nonCowLiver_nonPigLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_inRatLiverLoose_inCowLiverLoose_inPigLiverLoose_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueGenome/rheMac8.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 319
# Negatives: 54
# Performance: loss: 0.695058, accuracy: 0.651475, sensitivity: 0.626959, specificity: 0.796296, auc: 0.774324, precision: 0.947867, NPV: 0.265432, auprc: 0.462808, aunpvspec: 0.943320, auprg: 0.829821, F1: 0.754717, recallAt0.05FDR: 0.111111, recallAt0.1FDR: 0.120370
# Performance on non-restrictive laurasiatheria-specific enhancers and non-restrictive laurasiatheria-specific non-enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_inPigLiverLoose_nonMouseLiver_nonRatLiver_nonMacaqueLiver_test.bed.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllNonLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_inRatLiverLoose_inMacaqueLiverLoose_nonPigLiver_test.bed.gz --sequenceLength 500 --genomeFileName /home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/bosTau2015/GCF_000003205.7_Btau_5.0.1_genomic_withChrM.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 77
# Negatives: 86
# Performance: loss: 0.495026, accuracy: 0.788344, sensitivity: 0.740260, specificity: 0.831395, auc: 0.848686, precision: 0.797203, NPV: 0.781421, auprc: 0.825578, aunpvspec: 0.838509, auprg: 0.737058, F1: 0.767677, recallAt0.05FDR: 0.090909, recallAt0.1FDR: 0.259740
# Performance on shared liver and cortex enhancers and cortex enhancers that are not liver enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort_test_inCortex_summitPlusMinus250bp_andMacaqueLiverCortex_andRatLiverCortex.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_noRepeats_test_nonLiver_summitPlusMinus250bp_andMacaqueCortexNonLiver_andRatCortexNonLiver.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2892
# Negatives: 12357
# Performance: loss: 0.282232, accuracy: 0.897764, sensitivity: 0.791494, specificity: 0.922635, auc: 0.930461, precision: 0.705393, NPV: 0.949767, auprc: 0.809990, aunpvspec: 0.978801, auprg: 0.961863, F1: 0.745967, recallAt0.05FDR: 0.207988, recallAt0.1FDR: 0.442600
# Performance on mouse enhancers
gzip /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test.bed
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.narrowPeak_inLiuAll_nonCDS_enhancerShort.narrowPeak.gz --negNarrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test.bed.gz --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseGenome/mm10.fa --chroms chr1 --chroms chr2 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 3776
# Negatives: 3698
# Performance: loss: 0.454369, accuracy: 0.793083, sensitivity: 0.725636, specificity: 0.861952, auc: 0.877031, precision: 0.842947, NPV: 0.754706, auprc: 0.860741, aunpvspec: 0.884505, auprg: 0.777956, F1: 0.779905, recallAt0.05FDR: 0.069632, recallAt0.1FDR: 0.456733
# Performance on rat enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2576
# Negatives: 4405
# Performance: loss: 0.432676, accuracy: 0.801318, sensitivity: 0.792702, specificity: 0.806356, auc: 0.878984, precision: 0.705354, NPV: 0.869310, auprc: 0.805239, aunpvspec: 0.922341, auprg: 0.829527, F1: 0.746481, recallAt0.05FDR: 0.115101, recallAt0.1FDR: 0.263781
# Performance on macaque enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 2624
# Negatives: 3477
# Performance: loss: 0.426596, accuracy: 0.808474, sensitivity: 0.704078, specificity: 0.887259, auc: 0.886317, precision: 0.824961, NPV: 0.798912, auprc: 0.851275, aunpvspec: 0.909260, auprg: 0.825270, F1: 0.759741, recallAt0.05FDR: 0.239710, recallAt0.1FDR: 0.438072
# Performance on cow enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1664
# Negatives: 3721
# Performance: loss: 0.354496, accuracy: 0.845218, sensitivity: 0.798978, specificity: 0.865896, auc: 0.915308, precision: 0.727099, NPV: 0.905947, auprc: 0.823076, aunpvspec: 0.959666, auprg: 0.903061, F1: 0.761346, recallAt0.05FDR: 0.080529, recallAt0.1FDR: 0.365685
# Performance on pig enhancers
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Positives: 1627
# Negatives: 3057
# Performance: loss: 0.401538, accuracy: 0.815115, sensitivity: 0.725569, specificity: 0.862774, auc: 0.888577, precision: 0.737812, NPV: 0.855220, auprc: 0.790329, aunpvspec: 0.939165, auprg: 0.847802, F1: 0.731639, recallAt0.05FDR: 0.007068, recallAt0.1FDR: 0.100184
# Performance on hepatocyte enhancers
# Positives: 2516
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/wgEncodeAwgDnaseDukeHepatocytesUniPk_withPeakNames_hg38_summitExtendedMin50Max2XProtect5_nonCDS_enhancerShort_test.bed.gz --label 1 --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/hg38.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning # Accuracy: 0.188394
# Performance on reprocessed hepatocyte enhancers
# Positives: 563
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/liverEnhancerVsOtherSpeciesEnhMouseMacaqueRatCowPigPeakLessThan1kb_data_train.KerasModels/liver_mouseMacaqueRatCowPig_otherSpeciesEnhLooseNeg_500bp_conv5AllFiltNoInterPoolVeryHighMomL2SmallBatchPretrainBal.hdf5 --narrowPeakFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanDNase/Hepatocyte/atac/cb929f98-9900-4455-ad6a-48853584c300/call-reproducibility_idr/execution/idr.optimal_peak_nonCDS_enhancerShort.bed.gz --label 1 --sequenceLength 500 --genomeFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/HumanGenome/hg38.fa --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning # Accuracy: 0.692562

# Get reverse complement of motor cortex positive test set for each species
# Mouse
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Macaque
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Bat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa

# Get reverse complement of motor cortex negative test set for each species
# Mouse
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_RC.fa
# Rat
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp_RC.fa
# Macaque
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp_RC.fa
# Bat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_RefSeqNames_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_RC.fa

# Get reverse complement of liver positive test set for each species
# Mouse
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Rat
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Macaque
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Cow
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa
# Pig
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC.fa

# Get reverse complement of liver negative test set for each species
# Mouse
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_mm10Fixed_summitExtendedMin50Max2Protect5_nonMouseLiver_andRat_andCow_andPig_test_summitPlusMinus250bp_RC.fa
# Rat
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rn6Fixed_summitExtendedMin50Max2XProtect5_nonRatLiver_test_andMacaque_andCow_andPig_summitPlusMinus250bp_RC.fa
# Macaque
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_rheMac8Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonMacaqueLiver_test_andRat_andCow_andPig_summitPlusMinus250bp_RC.fa
# Cow
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_bosTau2015Fixed_summitExtendedMin50Max2XProtect5_RefSeqNames_nonCowLiver_test_andMacaque_andRat_andPig_summitPlusMinus250bp_RC.fa
# Pig
awk '/^>/{f=!d[$1];d[$1]=1}f' /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp.fa > /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp_unique.fa
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/reverseComplementFasta.py --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp_unique.fa --outputFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAllLoose_nonCDS_enhancerShort_susScr3Fixed_summitExtendedMin50Max2XProtect5_UCSCNames_nonPigLiver_test_andMacaque_andRat_andCow_summitPlusMinus250bp_RC.fa

# Get the Beluga Gliobla prediction accuracy for the motor cortex test set
# Mouse, positives
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/83b58f46-86db-4b66-a423-18a0f3a5d936_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6cbcff57-95ac-415b-b402-dbf3a4980612_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $9 >= 0.5' | grep -v DNase | wc # 990
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/83b58f46-86db-4b66-a423-18a0f3a5d936_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6cbcff57-95ac-415b-b402-dbf3a4980612_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 15924
# Accuracy: 0.06217030896
# Rat
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/b736fb7f-c337-403b-bf69-00e54c75c006_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/4cdc2356-4ce8-4140-aac9-2a1bdb681625_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $9 >= 0.5' | grep -v DNase | wc # 1110
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/b736fb7f-c337-403b-bf69-00e54c75c006_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/4cdc2356-4ce8-4140-aac9-2a1bdb681625_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 14170
# Accuracy: 0.07833450952
# Macaque
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/5690dcf2-e340-4b9f-b11b-bab2df3414fe_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6e884036-2f22-4e45-a023-f710532d12cc_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $9 >= 0.5' | grep -v DNase | wc # 740
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/5690dcf2-e340-4b9f-b11b-bab2df3414fe_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6e884036-2f22-4e45-a023-f710532d12cc_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 7524
# Accuracy: 0.09835194045
# Bat
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/0ec2506d-9700-4f84-85f1-8aee3a6fe6aa_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e39abe45-729b-4129-a79d-e57b828fe4e6_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $9 >= 0.5' | grep -v DNase | wc # 392
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/0ec2506d-9700-4f84-85f1-8aee3a6fe6aa_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e39abe45-729b-4129-a79d-e57b828fe4e6_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 4022
# Accuracy: 0.09746394828
# Full test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/evaluateSequencePredictions.py --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/83b58f46-86db-4b66-a423-18a0f3a5d936_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6cbcff57-95ac-415b-b402-dbf3a4980612_Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/b736fb7f-c337-403b-bf69-00e54c75c006_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/M1_AllReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/4cdc2356-4ce8-4140-aac9-2a1bdb681625_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/5690dcf2-e340-4b9f-b11b-bab2df3414fe_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6e884036-2f22-4e45-a023-f710532d12cc_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/0ec2506d-9700-4f84-85f1-8aee3a6fe6aa_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e39abe45-729b-4129-a79d-e57b828fe4e6_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posHeaderNum 1 --posPredCol 8 --posPredCol 8 --posPredCol 8 --posPredCol 8 --posPredCol 8 --posPredCol 8 --posPredCol 8 --posPredCol 8 --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/5690dcf2-e340-4b9f-b11b-bab2df3414fe_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/6e884036-2f22-4e45-a023-f710532d12cc_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_BelugaResults/4e1d5f5f-6db4-4542-8d42-3fbfeda2d70b_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_RC_BelugaResults/0f927249-fc7c-480b-b697-a762daf73729_nonCDS_enhancerShort_rn6_summitExtendedMin50Max2XProtect5_nonRatCortex_test_andMacaque_andBat_summitPlusMinus250bp_RC_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_nonMacaqueCortex_test_andRat_andBat_BelugaResults/dffdedb2-f517-4ea2-87ab-69e095aa844f_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_nonMacaqueCortex_test_andRat_andBat_RC_BelugaResults/1e0ee7b9-aaf9-4009-aecd-5f5e2fbe5ff6_nonCDS_enhancerShort_rheMac8_summitExtendedMin50Max2XProtect5_nonMacaqueCortex_test_andRat_andBat_summitPlusMinus250bp_RC_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_BelugaResults/3876109e-79ec-4a4a-9e9e-f2bdda37bb22_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_RC_BelugaResults/3fe16cd4-60cf-4eb9-96c0-f7b5d053b088_nonCDS_enhancerShort_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_RC_predictions.tsv --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negHeaderNum 1 --negPredCol 8 --negPredCol 8 --negPredCol 8 --negPredCol 8 --negPredCol 8 --negPredCol 8 --negPredCol 8 --negPredCol 8
# Performance: loss: 1.346418, accuracy: 0.520280, sensitivity: 0.077185, specificity: 0.977404, auc: 0.685085, precision: 0.778963, NPV: 0.506575, auprc: 0.678524, aunpvspec: 0.668674, F1: 0.140454, recallAt0.05FDR: 0.008771, recallAt0.1FDR: 0.027798
# Bat data only
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/evaluateSequencePredictions.py --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/0ec2506d-9700-4f84-85f1-8aee3a6fe6aa_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv --posPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/BatAtac/OfMBat1K/call-reproducibility_idr/execution/idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e39abe45-729b-4129-a79d-e57b828fe4e6_idr.optimal_peak.inM1_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv --posHeaderNum 1 --posHeaderNum 1 --posPredCol 8 --posPredCol 8 --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_BelugaResults/3876109e-79ec-4a4a-9e9e-f2bdda37bb22_nonCDS_enhancerShort_200MBat_summitExtendedMin50Max2XProtect5_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_predictions.tsv --negPredFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/nonCDS_enhancerShort_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_RC_BelugaResults/3fe16cd4-60cf-4eb9-96c0-f7b5d053b088_nonCDS_enhancerShort_HLrouAeg4_nonBatCortex_test_andMacaque_andRat_summitPlusMinus250bp_RC_predictions.tsv --negHeaderNum 1 --negHeaderNum 1 --negPredCol 8 --negPredCol 8
# Performance: loss: 0.732218, accuracy: 0.734851, sensitivity: 0.097464, specificity: 0.997135, auc: 0.793380, precision: 0.933333, NPV: 0.728618, auprc: 0.660483, aunpvspec: 0.890102, F1: 0.176497, recallAt0.05FDR: 0.049229, recallAt0.1FDR: 0.158130

# Get the brain model performance on the test set
python /home/ikaplow/RegulatoryElementEvolutionProject/src/RegElEvoCode/predictNewSequences.py --architectureFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/brainEnhancerVsOtherEuarchontaglireEnhMouseHumanMacaqueRatPeakLessThan1kb_data_train.KerasModels/hybrid_brainEnhancerShort_humanMouseMacaqueRat_euarchontaglireEnhLooseNeg_500bp_conv5MostFiltNoInterPoolVeryHighMomHighLRL2SmallBatchPretrainBal_architecture.json --weightsFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/brainEnhancerVsOtherEuarchontaglireEnhMouseHumanMacaqueRatPeakLessThan1kb_data_train.KerasModels/hybrid_brainEnhancerShort_humanMouseMacaqueRat_euarchontaglireEnhLooseNeg_500bp_conv5MostFiltNoInterPoolVeryHighMomHighLRL2SmallBatchPretrainBal.hdf5 --fastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/Cortex_All_ATAC_out/peak/macs2/idr/optimal_set/Pfenning_bulk_Ctx_nonCDS_enhancerShort_test_summitPlusMinus250bp_andMacaque_andRat_andBat.fa --negFastaFileName /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/OfM/peak/idr_reproducibility/idr.optimal_peak.inM1Loose_nonCDS_enhancerShort_GenBankNames_mm10_summitExtendedMin50Max2XProtect5_nonMouseCortex_test_andRat_andBat_summitPlusMinus250bp_andMacaqueNeg_andRatNeg_andBatNeg.fa --sequenceLength 500 --RC --classification --path /home/ikaplow/RegulatoryElementEvolutionProject/src/deeplearning
# Performance: loss: 0.531167, accuracy: 0.745717, sensitivity: 0.596614, specificity: 0.889349, auc: 0.852978, precision: 0.838554, NPV: 0.695927, auprc: 0.842491, aunpvspec: 0.859099, F1: 0.697191, recallAt0.05FDR: 0.178026, recallAt0.1FDR: 0.399640

# Get the Beluga Hepatocyte prediction accuracy for the liver test set
# Mouse
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/348f21e6-4ee5-4d9d-9af3-cb2819da6af8_idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/1b8ce15e-ea8e-4e34-aad2-6d75e3385fb6_idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $18 >= 0.5' | grep -v DNase | wc # 210
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/348f21e6-4ee5-4d9d-9af3-cb2819da6af8_idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/68eb27d1-23c7-4d42-80f9-b3dd46f5e18c/call-reproducibility_idr/execution/idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/1b8ce15e-ea8e-4e34-aad2-6d75e3385fb6_idr.optimal_peak.inLiuAll_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 7552
# Accuracy: 0.02780720338
# Rat
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/38bca0f4-4153-4f23-a492-de39adbd4055_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e4964111-95d8-4255-975e-05726390cd73_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $18 >= 0.5' | grep -v DNase | wc # 182
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/38bca0f4-4153-4f23-a492-de39adbd4055_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/RatAtac/Liver_GoodReps/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/e4964111-95d8-4255-975e-05726390cd73_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 5152
# Accuracy: 0.03532608695
# Macaque
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/68eab521-cb65-4d4d-9684-421bc197c779_optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/7157347f-b8af-4ac1-b4a9-9da5d21b285b_optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $18 >= 0.5' | grep -v DNase | wc # 160
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/68eab521-cb65-4d4d-9684-421bc197c779_optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MacaqueAtac/Liver/cromwell-executions/atac/1f5fd9f0-d389-4184-99cb-c1125fd7f064/call-reproducibility_idr/execution/optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/7157347f-b8af-4ac1-b4a9-9da5d21b285b_optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 5248
# Accuracy: 0.03048780487
# Cow
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/803760cb-cda8-412e-8d99-e67e904aec2e_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/745ac934-97c5-48cc-817a-8f18699bccac_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $18 >= 0.5' | grep -v DNase | wc # 218
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/803760cb-cda8-412e-8d99-e67e904aec2e_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/CowDNase/Liver/atac/888ae501-7469-4afa-8890-e36f14f6c83b/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/745ac934-97c5-48cc-817a-8f18699bccac_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 3328
# Accuracy: 0.06550480769
# Pig
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/f1a2cfdf-5549-43c6-9e24-c8c937597f67_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/9ad101da-e064-4049-935d-57628ec7aabd_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | awk 'BEGIN{OFS="\t"} $18 >= 0.5' | grep -v DNase | wc # 180
cat /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_BelugaResults/f1a2cfdf-5549-43c6-9e24-c8c937597f67_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_predictions.tsv /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/PigDNase/Liver/atac/568838bc-4846-4491-8787-45b51e65bfee/peak/idr_reproducibility/idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_BelugaResults/9ad101da-e064-4049-935d-57628ec7aabd_idr.optimal_peak_nonCDS_enhancerShort_test_summitPlusMinus250bp_RC_predictions.tsv | grep -v DNase | wc # 3254
# Accuracy: 0.05531653349